sw/device/tests/kmac_entropy_test.c

```diff
@@ -0,0 +1,281 @@
+
+ * @param kmac_p A KMAC handle.
+ * @param ret_code The return code from the most recent DIF call.
+ * @param result The boolean result (true = timeout happened).
+ * @return The result of the operation.
+ */
+static dif_result_t is_edn_timeout(dif_kmac_t *kmac_p, dif_result_t ret_code,
+                                   bool *result) {
+  dif_kmac_error_t err_reg;
+  dif_result_t ret_err = dif_kmac_get_error(kmac_p, &err_reg);
+  LOG_INFO("err_code = %08X", err_reg);
+  *result =
+      (ret_code == kDifError) && (kDifErrorEntropyWaitTimerExpired == err_reg);
+  return ret_err;
+}
+
+/**
+ * Helper function to print CSR.
+ *
+ * @param kmac_p A KMAC handle.
+ */
+static void print_csr(dif_kmac_t *kmac_p) {
+  uint32_t cfg_reg =
+      mmio_region_read32(kmac_p->base_addr, KMAC_CFG_SHADOWED_REG_OFFSET);
+  LOG_INFO("cfg_reg = %08X", cfg_reg);
+  uint32_t status_reg =
+      mmio_region_read32(kmac_p->base_addr, KMAC_STATUS_REG_OFFSET);
+  LOG_INFO("status_reg = %08X", status_reg);
+  uint32_t err_reg =
+      mmio_region_read32(kmac_p->base_addr, KMAC_ERR_CODE_REG_OFFSET);
+  LOG_INFO("err_reg = %08X", err_reg);
+  uint32_t cfg_regwen =
+      mmio_region_read32(kmac_p->base_addr, KMAC_CFG_REGWEN_REG_OFFSET);
+  LOG_INFO("cfg_regwen = %08X", cfg_regwen);
+  uint32_t entropy_period =
+      mmio_region_read32(kmac_p->base_addr, KMAC_ENTROPY_PERIOD_REG_OFFSET);
+  LOG_INFO("entropy_period = %08X", entropy_period);
+}
+
+bool test_main(void) {
+  LOG_INFO("Running KMAC ENTROPY test...");
+
+  // Initialize KMAC HWIP.
+  dif_kmac_t kmac;
+  dif_kmac_operation_state_t kmac_operation_state;
+  dif_kmac_error_t err_reg;
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC to use EDN entropy
+  // Set max values for the EDN timeout
+  dif_kmac_config_t config = (dif_kmac_config_t){
+      .entropy_mode = kDifKmacEntropyModeEdn,
+      .entropy_fast_process = false,
+      .entropy_hash_threshold = 0x0,
+      .entropy_wait_timer = 0x0020,
+      .entropy_prescaler = 0x0001,
+      .message_big_endian = false,
+      .output_big_endian = false,
+      .sideload = false,
+      .msg_mask = false,
+  };
+
+  bool is_kmac_locked;
+  bool expected_timeout_status;
+  bool observed_timeout_status;
+  kmac_edn_timeout_t timeout_test;
+  size_t i = 0;
+
+  // Handle string encoding
+  dif_kmac_customization_string_t str_buffer;
+  CHECK_DIF_OK(dif_kmac_customization_string_init(
+      kmac_test_vector.customization_string,
+      kmac_test_vector.customization_string_len, &str_buffer));
+
+  // When customization_string is empty, use NULL to activate empty str path
+  dif_kmac_customization_string_t *s_buff =
+      kmac_test_vector.customization_string_len == 0 ? NULL : &str_buffer;
+
+  size_t digest_l =
+      kmac_test_vector.digest_len_is_fixed ? kmac_test_vector.digest_len : 0;
+  uint32_t out[DIGEST_LEN_KMAC_MAX];
+
+  for (; i < nr_of_tests; i++) {
```

Comment:
nit: why not also declare the `i` here, so that it's a one-liner
```suggestion
  for (size_t i = 0; i < nr_of_tests; i++) {
```

Reply:
Thanks, I fixed it.